src/Symfony/Component/Security/Http/Firewall.php

```diff
@@ -122,6 +123,10 @@ public static function getSubscribedEvents()
     protected function callListeners(RequestEvent $event, iterable $listeners)
     {
         foreach ($listeners as $listener) {
+            if (!$listener instanceof FirewallListenerInterface) {
+                trigger_deprecation('symfony/security-http', '7.4', 'Using a callable as firewall listener is deprecated, extend "%s" or implement "%s" instead.', AbstractListener::class, FirewallListenerInterface::class);
+            }
+
             $listener($event);
```

Comment:
is this still fine given `__invoke()` is not specified by `FirewallListenerInterface`? Has it ever been or should it be `authenticate()`? Just wondering as I feel like something's been lost in my memory

Reply:
So any non-invokable userland listener is broken. should be fixed on 6.4

Reply:
Fixed by #60785